server/utils/cloudFunctions/functions/test/algoliaTests/algoliaIntegrationTests.test.js

```diff
@@ -89,154 +53,148 @@ const { DB } = require('../../firebaseConfig');
 const { DEV_ADS_INDEX } = 
     require('../../algoliaCloudFunctions/algoliaConfig');
 
+const { makeRandomID } = require('../makeRandomID');
+
+const { retryAssertions } = require('../retryAssertions');
+
 describe("Algolia integrations tests", () => {
   // After every 'describe' block, reset the test environments. 
-  // Since these are live dev DBs, this process can be very flaky.
-  // In general, don't expect an 'advertiser' will be reset from 
-  // test to test. 
   // Note: this will also delete ad records in the corresponding 
   // algolia index. 
   after(() => {
     deleteCollection(DB, 'dev_ads');
   });
-
+  
   describe("test_createRecord", () => {
-    it("should propogate an ad document into algolia index", function(done) {
-      const ad = {advertiser: "a", startDate: "2019-10-15"};
-
-      DEV_ADS_COLLECTION.doc("a").set(ad).then(() => {
-        setTimeout(function(){
-          DEV_ADS_INDEX.getObject("a").then(content => {
-            chai.expect(content.data).to.deep.equal(ad);
-          }).catch(err => console.log(err));
-
-          done();
-        }, TIMEOUT_10S);
-      });
+    it("should propogate an ad document into algolia index", async () => {
+      const ad = {advertiser: makeRandomID(), startDate: "2019-10-15"};
```

Comment:
define this as its own variable, to avoid the possibility of ad being modified in one of your methods, thus changing this ID.